06week/higherOrder.js

```diff
@@ -2,102 +2,76 @@
 
 const assert = require('assert');
 
-function forEach(arr, callback) {
-  // Your code here
-}
+
+const arr=[10,20,30];
 
 function map(arr, callback) {
-  // Your code here
+  let newArr=[];
+  for (let i=0; i<arr.length; i++){
+    let elements=callback(arr[i]);
+    newArr.push(elements);
+  }
+  return newArr;
 }
 
 function filter(arr, callback) {
-  // Your code here
-}
+  const filterArr=[];
+  for (let i=0; i<arr.length; i++){
+    const result = calllback(arr[i], i, arr);
+    if (result)
+    filterArr.push(arr[i]);
+  }
 
-function some(arr, callback) {
-  // Your code here
+ return filterArr;
 }
 
-function every(arr, callback) {
-  // Your code here
-}
+function reduce(array, callback, accumulator) {
```

Comment:
Just remember to indent your code, it looks like this reduce function could use some indentation